scripts/source-maintenance.sh

```diff
@@ -21,6 +21,38 @@
 # If code alteration takes place the process is halted for manual intervention.
 #
 
+# whether to continue execution after a failure
+# TODO: Expand the subset of failures covered by this feature; see run_().
+KeepGoing="no"
+# the actual name of the directive that enabled keep-going mode
+KeepGoingDirective=""
+
+# command-line options
+while [ $# -ge 1 ]; do
+    case "$1" in
+    --keep-going|-k)
+        KeepGoing=yes
+        KeepGoingDirective=$1
+        shift
+        ;;
+    *)
+        echo "Usage: $0 [--keep-going|-k]"
+        echo "Unsupported command-line option: $1"
+        exit 1;
+        ;;
+    esac
+done
+
+# an error code seen by a KeepGoing-aware command (or zero)
+SeenErrors=0
+
+
+if ! git diff --quiet; then
+	echo "There are unstaged changes. This script may modify sources."
+	echo "Stage changes to avoid permanent losses when things go bad."
+	exit 1
+fi
+
 # On squid-cache.org we have to use the python scripted md5sum
```

Comment:
This whole squid-cache.org block is rotten and is by now unnecessary. In fact, it hasn't worked for years.

Reply:
I know.  It is out of scope though for this feature change.